components/builder-http-gateway/src/http/middleware.rs

```diff
@@ -246,7 +227,32 @@ impl AfterMiddleware for Cors {
     }
 }
 
-pub fn session_create_github(req: &mut Request, token: String) -> IronResult<Session> {
+pub fn session_create(
+    req: &mut Request,
+    features: FeatureFlags,
+    token: SessionToken,
+) -> IronResult<Session> {
+    let mut request = SessionGet::new();
+    request.set_token(token);
+    let conn = req.extensions.get_mut::<XRouteClient>().unwrap();
+    match conn.route::<SessionGet, Session>(&request) {
+        Ok(session) => {
+            let flags = FeatureFlags::from_bits(session.get_flags()).unwrap();
+            if !flags.contains(features) {
+                let err = NetError::new(ErrCode::ACCESS_DENIED, "net:auth:feature-flags");
+                return Err(IronError::new(err, Status::Forbidden));
+            }
+            Ok(session)
+        }
+        Err(err) => {
+            let status = net_err_to_http(err.get_code());
+            let body = itry!(serde_json::to_string(&err));
+            Err(IronError::new(err, (body, status)))
+        }
+    }
+}
```

Comment:
I'm confused - this is the `session_create` function, but all we try to do is fetch a session and if one doesn't already exist, return an error?  Shouldn't we be routing a `SessionCreate` message?

Reply:
Ah, the name isn't quite right.. the it's really session_validate - I'll fix it. The session is actually getting created elsewhere (either in the builder api oauth flow, or in the session_create_github for the cli auth flow).

Reply:
Perfect.